### PR TITLE
Fix issue 172

### PR DESCRIPTION
**Pull request by mtimmerm**
Fix issue 172 -- erroneous implementation of double-check locking caused NPE in clients of LazyDeserializer.

### CODE REVIEW

**Comment by dblock** [comment]
This needs a test, please?

File: java-client/src/test/java/org/opensearch/client/opensearch/json/JsonpMapperTest.java
Comment (dblock) [inline java-client/src/test/java/org/opensearch/client/opensearch/json/JsonpMapperTest.java:162]: Does it really take 5 seconds? Doubt it, maybe this should be very short to avoid hanging the test unnecessarily?

**Review by dblock — APPROVED** [review]
LGTM, will wait for an answer/update for below before merging

File: java-client/src/test/java/org/opensearch/client/opensearch/json/JsonpMapperTest.java
Comment (dblock) [reply java-client/src/test/java/org/opensearch/client/opensearch/json/JsonpMapperTest.java:162]: Oh 🤦 long day :)

**Comment by ecastilloseekr** [comment]
Hello, thank you for this fix! Do you guys plan to create a tag with this fix anytime soon? Thank you.  @mtimmermansTa @Arsnael

**Comment by dblock** [comment]
We have #174 open, working in it, but no immediate ETA, cc: @bbarani @gaiksaya.

**Pull request merged**
